ccd/models/lasso.py

```diff
@@ -29,17 +29,26 @@ def coefficient_matrix(dates, avg_days_yr, num_coefficients):
 
     matrix = np.zeros(shape=(len(dates), 7), order='F')
 
+    # lookup optimizations
+    # Before optimization - 12.53% of total runtime
+    # After optimization  - 10.57% of total runtime
+    w12 = w * dates
+    w34 = 2 * w12
+    w56 = 3 * w12
+    cos = np.cos
+    sin = np.sin
+
     matrix[:, 0] = dates
-    matrix[:, 1] = np.cos(w * dates)
-    matrix[:, 2] = np.sin(w * dates)
+    matrix[:, 1] = cos(w12)
+    matrix[:, 2] = sin(w12)
 
     if num_coefficients >= 6:
-        matrix[:, 3] = np.cos(2 * w * dates)
-        matrix[:, 4] = np.sin(2 * w * dates)
+        matrix[:, 3] = cos(w34)
+        matrix[:, 4] = sin(w34)
 
     if num_coefficients >= 8:
```

Comment:
Did you want to go ahead and bundle a fix for the "if" logic, making it better as well?